assets/js/components/Dashboard/AlertDetails.tsx

```diff
@@ -19,56 +25,89 @@ const AlertDetails: ComponentType = () => {
-      } else {
-        setSelectedAlert(selectedAlert);
-      }
+    const selectedAlert = screenplayContext.alerts.length
+      ? screenplayContext.alerts.find((alert) => alert.id === id)
+      : null;
+
+    if (selectedAlert) {
+      setContextState(screenplayContext);
+      setSelectedAlert(selectedAlert);
+      setShow(false);
+    } else {
+      setShow(true);
     }
-  }, [alerts]);
+  }, [screenplayContext]);
 
   return selectedAlert ? (
-    <div className="alert-details">
-      <div className="page-content__header">
-        <div>
-          <Button
-            className="alert-details__back-button"
-            data-testid="alert-details-back-button"
-            onClick={() => navigate("/alerts", { replace: true })}
-          >
-            <ArrowLeft /> Back
-          </Button>
-          <span>
-            {formatEffect(selectedAlert.effect)} #{selectedAlert.id}
-          </span>
-          <Button
-            href={alertsUiUrl + `/edit/${selectedAlert.id}`}
-            target="_blank"
-            className="alert-details__external-link"
-          >
-            Edit Alert <ArrowUpRight />
-          </Button>
+    <>
+      <div className="alert-details">
+        <div className="page-content__header">
+          <div>
+            <Button
+              className="alert-details__back-button"
+              data-testid="alert-details-back-button"
+              onClick={() => navigate("/alerts", { replace: true })}
+            >
+              <ArrowLeft /> Back
+            </Button>
+            <span>
+              {formatEffect(selectedAlert.effect)} #{selectedAlert.id}
+            </span>
+            <Button
+              href={alertsUiUrl + `/edit/${selectedAlert.id}`}
+              target="_blank"
+              className="alert-details__external-link"
+            >
+              Edit Alert <ArrowUpRight />
+            </Button>
+          </div>
         </div>
+        {Object.keys(screensByAlertMap).length !== 0 && (
+          <div className="page-content__body">
+            <AlertCard alert={selectedAlert} classNames="selected-alert" />
+            <PlacesList
+              places={placesWithSelectedAlert(
+                selectedAlert,
+                places,
+                screensByAlertMap
+              )}
+              noModeFilter
+              isAlertPlacesList
+            />
+          </div>
+        )}
       </div>
-      <div className="page-content__body">
-        <>
-          <AlertCard alert={selectedAlert} classNames="selected-alert" />
-          <PlacesList
-            places={placesWithSelectedAlert(
-              selectedAlert,
-              places,
-              screensByAlertMap
-            )}
-            noModeFilter
-            isAlertPlacesList
-          />
-        </>
-      </div>
-    </div>
+      <Modal
+        className="alert-not-found"
+        backdropClassName="alert-not-found"
+        show={show}
+        onHide={handleClose}
```

Comment:
If a user closes the modal instead of clicking the "Go to Posted Alerts" button, do they just see some sort of empty alert detail page? (That's fine, just trying to make sure I understand the markup/the component logic)

Reply:
Closing the modal by simply clicking off the modal will just remove the modal layer, without updating the alert page from its previous state; the modal will appear again in a few seconds when the data fetch happens again. I allowed that workflow because I thought it might be interesting to let the user glance at the page again before navigating away, but I'm not sure that's normal or expected. I'll think about it for a moment...

Reply:
Yeah, I think I'll scrap it, because leaving it open was not a workflow that design team made